example/segnet/common/contrib_metrics.py

```diff
@@ -0,0 +1,156 @@
+    >>> acc = mx.metric.Accuracy()
+    >>> acc.update(preds = predicts, labels = labels)
+    >>> print acc.get()
+    ('accuracy', 0.6666666666666666)
+    """
+    def __init__(self, axis=1, name='accuracy',
+                 output_names=None, label_names=None, ignore_label=-1):
+        super(Accuracy, self).__init__(
+            name, axis=axis,
+            output_names=output_names, label_names=label_names)
+        self.axis = axis
+        self.ignore_label = ignore_label
+
+    def update(self, labels, preds):
+        """Updates the internal evaluation result.
+
+        Parameters
+        ----------
+        labels : list of `NDArray`
+            The labels of the data.
+
+        preds : list of `NDArray`
+            Predicted values.
+        """
+        mx.metric.check_label_shapes(labels, preds)
+
+        for label, pred_label in zip(labels, preds):
+            if pred_label.shape != label.shape:
+                pred_label = mx.ndarray.argmax(pred_label, axis=self.axis)
+            pred_label = pred_label.asnumpy().astype('int32')
+            label = label.asnumpy().astype('int32')
+
+            mx.metric.check_label_shapes(label, pred_label)
+
+            self.sum_metric += (pred_label.flat == label.flat).sum()
+            self.num_inst -= len(label[label == self.ignore_label])
+            self.num_inst += len(pred_label.flat)
+
+class CrossEntropy(mx.metric.EvalMetric):
+    """Computes Cross Entropy loss.
+
+    The cross entropy over a batch of sample size :math:`N` is given by
+
+    .. math::
+       -\\sum_{n=1}^{N}\\sum_{k=1}^{K}t_{nk}\\log (y_{nk}),
+
+    where :math:`t_{nk}=1` if and only if sample :math:`n` belongs to class :math:`k`.
+    :math:`y_{nk}` denotes the probability of sample :math:`n` belonging to
+    class :math:`k`.
+
+    Parameters
+    ----------
+    eps : float
+        Cross Entropy loss is undefined for predicted value is 0 or 1,
+        so predicted values are added with the small constant.
+    name : str
+        Name of this metric instance for display.
+    output_names : list of str, or None
+        Name of predictions that should be used when updating with update_dict.
+        By default include all predictions.
+    label_names : list of str, or None
+        Name of labels that should be used when updating with update_dict.
+        By default include all labels.
+
+    Examples
+    --------
+    >>> predicts = [mx.nd.array([[0.3, 0.7], [0, 1.], [0.4, 0.6]])]
+    >>> labels   = [mx.nd.array([0, 1, 1])]
+    >>> ce = mx.metric.CrossEntropy()
+    >>> ce.update(labels, predicts)
+    >>> print ce.get()
+    ('cross-entropy', 0.57159948348999023)
+    """
+    def __init__(self, eps=1e-12, name='cross-entropy',
+                 output_names=None, label_names=None):
+        super(CrossEntropy, self).__init__(
+            name, eps=eps,
+            output_names=output_names, label_names=label_names)
+        self.eps = eps
+
+    def update(self, labels, preds):

```

Comment:
Why not use the native one in metric.py?

Reply:
1.The native 'CorssEntropy' in metric.py can only used to pred_label with 2-d.
2. In this example, the pred_label is 4-d and the softmax result located in axis 1. We must swap the softmax result to the last axis.
3. Because of the ignore_label=11, the max number of label is 11, but the size of softmax result is only 0-10. We must add one column data to use the command 
`prob = pred[numpy.arange(label.shape[0]), numpy.int64(label)]`